staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go

```diff
@@ -662,18 +668,49 @@ func (o *DiffOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []str
 		return err
 	}
 
-	if cmdutil.GetFlagBool(cmd, "prune") {
+	if o.Prune {
 		mapper, err := f.ToRESTMapper()
+		o.tracker = newTracker()
+
 		if err != nil {
 			return err
 		}
 
-		resources, err := prune.ParseResources(mapper, cmdutil.GetFlagStringArray(cmd, "prune-allowlist"))
-		if err != nil {
-			return err
+		// Taken from apply.go
+		if o.ApplySetRef != "" {
+			parent, err := apply.ParseApplySetParentRef(o.ApplySetRef, mapper)
+
+			if err != nil {
+				return fmt.Errorf("invalid parent reference %q: %w", o.ApplySetRef, err)
+			}
+
+			if o.EnforceNamespace && parent.IsNamespaced() {
+				parent.Namespace = o.CmdNamespace
+			}
+
+			tooling := apply.ApplySetTooling{Name: "kubectl", Version: apply.ApplySetToolVersion}
+			restClient, err := f.UnstructuredClientForMapping(parent.RESTMapping)
+			if err != nil {
+				return fmt.Errorf("failed to initialize RESTClient for ApplySet: %w", err)
+			}
```

Comment:
nit: add newline between if statements
```suggestion
			}
			
```